### ISSUE TITLE
Remove Index for `nix profile remove` and `nix profile upgrade`

### ISSUE THREAD

**Issue by iFreilicht** [NixOS/nix#9171]
This is a follow-up issue to #8678. After adding names as the primary method of addressing packages installed into profiles, indices are not useful anymore and just clutter the output of `nix profile list`.

**Describe the solution you'd like**
Remove indices as a mechanism for addressing packages.

**Describe alternatives you've considered**
We can leave them in. See below.

**Additional context**
In an effort to reduce frustration and gain UX insights, a deprecation warning is shown (once #8678 is merged) when users use an index to refer to a package when running `nix profile remove` or `nix profile upgrade`:

```
warning: Indices are deprecated and will be removed in a future version!
         Refer to packages by their `Name` as printed by `nix profile list`.
         If you want to vote against this change, please leave a comment here:
         https://github.com/NixOS/nix/issues/9171
```

If enough people leave comments on this issue about indices being a useful mechanism to address profile elements, we may want to leave them in.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by kolloch** [comment]
*Compatibility*

I would keep the indexes supported as long as possible so that current shell scripts continue to work.

I am really a fan of the new names, though, the current systems is hard to work with. But I do not want parts like this from our bootstrapping scripts fail suddenly:

```
                installed_idx=$(
                  ${nix}/bin/nix profile list --profile $profile --json |
                  ${jq}/bin/jq -r '.elements | to_entries[] | select((.value.originalUrl // "") | contains("nix-bootstrap") and contains("nexxiot")) | .key'
                )
                if [ -n "$installed_idx" ]; then
                  echo "Removing existing install in default profile..."
                  $sudo ${nix}/bin/nix profile remove --profile $profile $installed_idx;
                fi
```
Context: The current nix ecosystem encourages using experimental features -- because flakes and the new CLI have not yet been stabilized. That, unfortunately, means that the line between truly experimental features and assumed to be stable features is quite blurry. Let's not break things that our users already got working!

**Comment by iFreilicht** [comment]
Yes, you're right on this. We came to the same conclusion in the PR, and Indices will continue to work, but also print a warning. Maybe we can then remove them in a few years 😇

**Comment by iFreilicht** [comment]
Seems that message is already working before it was even released :D Thank you for your insights!

**Comment by bobvanderlinden** [comment]
I think this was resolved by https://github.com/NixOS/nix/pull/9656. Indexes were removed and named identifiers replaced them.

**Comment by iFreilicht** [comment]
Yup 👍
